HLTrigger/Egamma/plugins/HLTEgammaCombMassFilter.cc

```diff
@@ -43,7 +45,35 @@ void HLTEgammaCombMassFilter::fillDescriptions(edm::ConfigurationDescriptions& d
 bool HLTEgammaCombMassFilter::hltFilter(edm::Event& iEvent,
                                         const edm::EventSetup& iSetup,
                                         trigger::TriggerFilterObjectWithRefs& filterproduct) const {
-  //right, issue 1, we dont know if this is a TriggerElectron, TriggerPhoton, TriggerCluster (should never be a TriggerCluster btw as that implies the 4-vectors are not stored in AOD)
+
+  using namespace trigger;
+  if (saveTags()) {
+    filterproduct.addCollectionTag(l1EGTag_);
+  }
+
+  edm::Handle<trigger::TriggerFilterObjectWithRefs> PrevFilterOutput;
+  iEvent.getByToken(firstLegLastFilterToken_, PrevFilterOutput);
+
+  std::vector<edm::Ref<reco::RecoEcalCandidateCollection> > phoCandsPrev;
+  PrevFilterOutput->getObjects(trigger::TriggerPhoton, phoCandsPrev);
+  std::vector<edm::Ref<reco::RecoEcalCandidateCollection> > clusCandsPrev;
+  PrevFilterOutput->getObjects(trigger::TriggerCluster, clusCandsPrev);
+  std::vector<edm::Ref<reco::ElectronCollection> > eleCandsPrev;
+  PrevFilterOutput->getObjects(trigger::TriggerElectron, eleCandsPrev);
+
+  if (!phoCandsPrev.empty()) {
+    for (auto& phoCand : phoCandsPrev) {
+      filterproduct.addObject(trigger::TriggerPhoton, phoCand);
+    }
+  } else if (!clusCandsPrev.empty()) {
+    for (auto& clusCand : clusCandsPrev) {
+      filterproduct.addObject(trigger::TriggerCluster, clusCand);
+    }
+  } else if (!eleCandsPrev.empty()) {
+    for (auto& eleCand : eleCandsPrev) {
+      filterproduct.addObject(trigger::TriggerElectron, eleCand);
+    }
+  }
```

Comment:
Normally (but I guess there is no universal rule on this), the trigger objects are added for those candidates which pass some kinematic/trigger selection inside the `HLTFilter`, e.g. the candidates forming pairs passing the mass cut. Here, it seems that simply every candidate is saved regardless. Is this really intended ?

Reply:
Sure I will try to fix it, thanks for pointing this out

Reply:
The problem has been fixed in the latest commits.